pkg/services/sqlstore/migrations/annotation_mig.go

```diff
@@ -178,6 +178,10 @@ func addAnnotationMig(mg *Migrator) {
 	mg.AddMigration("Add index for alert_id on annotation table", NewAddIndexMigration(table, &Index{
 		Cols: []string{"alert_id"}, Type: IndexType,
 	}))
+
+	mg.AddMigration("Increase tags column to length 4096", NewRawSQLMigration("").
+		Postgres("ALTER TABLE annotation ALTER COLUMN tags TYPE VARCHAR(4096);").
+		Mysql("ALTER TABLE annotation MODIFY tags VARCHAR(4096);"))
```

Comment:
Ze juicy part.

This looks ok to me, but I'd appreciate a second pair of eyes on this or would like to set aside some time to benchmark it/browse MySQL/Postgres docs to figure out the risks of running this if you say have a few million annotations before I'm comfortable with approving this 🙈

In case anyone wonders, the reason why SQLite doesn't need a migration is because it has dynamic typing by default which we're using, so it doesn't actually care about limits on VARCHARs, see [datatypes in SQLite3](https://www.sqlite.org/datatype3.html)

> Note that numeric arguments in parentheses that following the type name (ex: "VARCHAR(255)") are ignored by SQLite - SQLite does not impose any length restrictions (other than the large global [SQLITE_MAX_LENGTH](https://www.sqlite.org/limits.html#max_length) limit) on the length of strings, BLOBs or numeric values.

Reply:
the SQL considerations are also discussed [here](https://github.com/grafana/grafana/discussions/51815)

I'll trying to monitor its performance but we have run [similar migrations](https://github.com/grafana/grafana/blob/17fc5d82d32c080dceeeec02837f7f09e2944d98/pkg/services/sqlstore/migrations/dashboard_mig.go#L94) for other tables

Reply:
I have created 500k annotations using generators following [this](https://github.com/grafana/grafana/pull/21915#issuecomment-582328787) and I was able to run the migration without issues.